### DISCUSSION TITLE
@click.outside listener on an undisplayed element responds to click that displays the element

### DISCUSSION THREAD

**Discussion by mtnorthrop** [alpinejs/alpine#1893] [7. Bugs (archived)] [answered]
Hello,

I have a `@click.outside` listener on an undisplayed element that's responding to the same click that causes the element to be displayed. 

I'm using Alpine v3, and I don't think I was experiencing this issue with the same code when using v2.

This should replicate the issue:

```html
<button x-data @click="$dispatch('test-open')">
    Open it
</button>

<div x-data="test" @test-open.window="showDiv()">
    <template x-if="open">
        <div style="width: 256px; height: 256px; background-color: red; color: white;" @click.outside="clickOutside()">
            I am open
        </div>
    </template>
</div>

<script type="text/javascript">
    document.addEventListener('alpine:init', () => {
        Alpine.data('test', () => ({
            open: false,

            showDiv() {
                this.open = true;
            },

            clickOutside() {
                console.log('Clicked outside');
            },
        }));
    })
</script>
```

The behavior I'm expecting is that clicking on the button will open the div, and clicking off the open div will log "Clicked outside". However, instead, I'm also seeing the "Clicked outside" log when I first click on the button that opens the div. (This is a simplified example; I'm actually working on building a lightbox component. The lightbox trigger is separate from the modal, so I'm using events on the window object to communicate between them.)

Is this a bug in Alpine v3, or am I approaching things incorrectly?

Thanks in advance!
Mark

**Comment by KevinBatdorf** [comment, marked as answer]
I think you might need `<button x-data @click.stop="$dispatch('test-open')">` to keep the event from bubbling up

**Reply by mtnorthrop** [reply]
Thank you! That was the problem, and adding `.stop` to `@click` fixed everything. I really appreciate your taking the time to respond!
